src/freenas/usr/local/sbin/disk_resize

```diff
@@ -121,6 +122,12 @@ ata|sata)
 	if [ -z "${size}" ]; then
 		size=${maxsect}
 	fi
+	echo "Waiting up to ${udevd_timeout} seconds for udevd events to settle"
+	udevadm settle -t ${udevd_timeout}
+	if [ $? -ne 0 ]; then
+		echo "Resize failed. Udevd events failed to settle within ${udevd_timeout} seconds."
+		exit 1
```

Comment:
I don't think you are required to fail here.  It may fail later if still hit the race, or it may succeed.  Warning message may be enough.

Reply:
Ah, right. Thank you. Updated.